indexer/run.sh

```diff
@@ -33,23 +36,100 @@ fetch --nohooks chromium
 
 cd src
 
+mkdir -p out/Default
+export BUILD_DIR=$(readlink -f out/Default)
+
+echo "target_os = [ 'linux', 'android', 'chromeos', 'fuchsia' ]" >> ../.gclient
+
+gclient sync
+
+gclient runhooks
+
+# Create a release, will be empty for now and incrementally populated
+# throughout the indexing pipeline.
+
+DATE=$(date -u +%Y%m%d)
+
+COMMIT=$(git rev-parse --short HEAD)
+
+gh release create $COMMIT --repo clangd/chrome-remote-index \
+  --title="Index at $DATE" \
+  --notes="Chromium index artifacts at $COMMIT with project root \`$PWD\`."
+
+# The indexing pipeline is common. Each platform will only have to do the
+# preparation step (set up the build configuration and install dependencies).
+
+# $1: the platform name.
+index() {
```

Comment:
nit: I'd move gn gen args bit into here as well by taking two arguments, e.g:

`index "chromeos" 'target_os="chromeos"'`

and index would look like:
```
index() {
PLATFORM = $1
GN_ARGS = $2

...

}
```